### ISSUE TITLE
Loading lots of data client-side

### ISSUE THREAD

**Issue by itjustwerks** [angular-ui/ui-grid#4309]
I have around 650 rows of data I want to render using UI Grid. However, only the first 6 rows show up. If I set virtualizationThreshold='651', the rows all load, but performance is mind-numbingly poor. Is there a solution that allows virtualization when dealing with large data sets? Please help!

**Comment by PaulL1** [comment]
Virtualisation should be automatic.  I'm assuming that you tried scrolling?  I'm a bit lost on what could be going on - do the tutorials work for you?  Can you create a plunker?

**Comment by itjustwerks** [comment]
Yes, I tried scrolling, and it's literally just the 6 rows. Inspect Element shows there are only 6 row divs rendered in the ng-repeat for rowContainer.renderedRows, even though I can display on the same page a plain ng-repeat for the bound data object and see 650 records. I'll try to put together a plunker later today.

**Comment by PaulL1** [comment]
It should only render a few rows because of virtualisation (whether 6 is right is a different question).  I think the plunker will be key to diagnose.

**Comment by itjustwerks** [comment]
Here's a plunk that replicates the issue exactly: http://plnkr.co/edit/LIS1SiwKqbUZIPZ0xORr?p=preview.

I use Firebase as my backend to retrieve the data, which has a $loaded promise after the request is complete, at which point I bind the data to the grid. Here are a couple of things to do to expose the really weird behavior:

1) Sort the columns, or in the "Deal Status" column, type "Prospect" in the filter. Both actions change the data in the 6 rows that appear, indicating that the larger data set is somehow behind there but not being rendered.

2) In the "Deal Status" column, type "Closed" in the filter. Now you will see all 20 Closed properties because the virtualizationThreshold defaults to 20. Changing that setting to '33' in script.js allows New, Closed and Mutual filters to work, but not Prospects. Changing it to '685' lets everything work, but oh so slowly, and scrolling when filtered to "Prospects" is really rough if it works at all.

I'm going to look into using paging as a solution, but I don't understand why this doesn't work as is?

Thanks!

**Comment by swalters** [comment]
seems to be a bug with flatEntityAccess.  Remove that setting and it works

**Comment by itjustwerks** [comment]
@swalters Interesting. Commenting out the flatEntityAccess setting does not change the behavior at all for me. Can anyone else confirm that fixes it for them?

**Comment by swalters** [comment]
Yes, I was just pasting this in:
http://plnkr.co/edit/3vlKoePordZRE4hEXp7E?p=preview

I think it may be a combination of things.

1. the grid height should only be set to a how many visible rows you want to see.
2. I removed several settings after changing the CSS.  The flatEntityAccess was the last I removed that made it work.
http://plnkr.co/edit/3vlKoePordZRE4hEXp7E?p=preview

**Comment by itjustwerks** [comment]
WOW! I played around with the settings further and I think it was actually the "rowHeight: '33'" line that caused the issue. I added everything else back in, including the dynamic height, and it completely works as needed now. Thank you so much!

http://plnkr.co/edit/3vlKoePordZRE4hEXp7E?p=preview

**Comment by itjustwerks** [comment]
Oops...I meant my plunker for final edits: http://plnkr.co/edit/LIS1SiwKqbUZIPZ0xORr?p=preview

**Comment by eMazeika** [comment]
Why this issue was closed as rows can be rendered correctly only if rowHeight is not set? I've the same issue and I need to have custom rowHeight.

**Comment by itjustwerks** [comment]
@eMazeika, good point. It's closed re: my particular use-case, but there definitely is something worth investigating in the rowHeight.
